tests/connect/standalone-x509-extract_username-001.phpt

```diff
@@ -0,0 +1,37 @@
+--TEST--
+Connect to MongoDB with SSL and X509 auth and username retrieved from cert
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; NEEDS("STANDALONE_X509"); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$SSL_DIR = realpath(__DIR__ . '/../../scripts/ssl/');
+
+$driverOptions = [
+    // libmongoc does not allow the hostname to be overridden as "server"
+    'allow_invalid_hostname' => true,
+    'weak_cert_validation' => false,
+    'ca_file' => $SSL_DIR . '/ca.pem',
+    'pem_file' => $SSL_DIR . '/client.pem',
+    // TODO: this doesn't appear to have any effect. Does the PEM file not have a password?
```

Comment:
Looking through [`mongoc_ssl_extract_subject()`](https://github.com/mongodb/mongo-c-driver/blob/1.4.0/src/mongoc/mongoc-ssl.c#L57), only `_mongoc_secure_transport_extract_subject()` appears to use the password. OpenSSL and Secure Channel implementations seem to ignore the parameter.

If I attempt to read the `client.pem` file with `ssh-keygen -y -f client.pem` (as documented [here](http://serverfault.com/a/628924)), I can see its contents without a password prompt. Therefore, I assume it was never created with a password. I'm going to remove the `pem_pwd` option from both of these tests.

@bjori: do you concur? Furthermore, can you confirm whether or not OpenSSL and Secure Channel support passwords on the PEM file?